Index: app/src/main/java/sg/edu/np/mad/pennywise/EditProfile.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/sg/edu/np/mad/pennywise/EditProfile.java b/app/src/main/java/sg/edu/np/mad/pennywise/EditProfile.java
--- a/app/src/main/java/sg/edu/np/mad/pennywise/EditProfile.java	
+++ b/app/src/main/java/sg/edu/np/mad/pennywise/EditProfile.java	
@@ -5,6 +5,7 @@
 import androidx.appcompat.app.AppCompatActivity;
 
 import android.content.Intent;
+import android.content.SharedPreferences;
 import android.os.Bundle;
 import android.text.TextUtils;
 import android.util.Log;
@@ -16,6 +17,11 @@
 
 public class EditProfile extends AppCompatActivity {
 
+
+    public String GLOBAL_PREFS = "myPrefs";
+    public String MY_EMAIL = "MyEmail";
+    public String MY_PASSWORD = "MyPassword";
+    SharedPreferences sharedPreferences;
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
@@ -24,6 +30,11 @@
         EditText editEmail = findViewById(R.id.editprofile_email);
         EditText editOldPassword = findViewById(R.id.editprofile_password);
         EditText editNewPassword = findViewById(R.id.editprofile_newpassword);
+        // Retrieve the username from the shared prefs
+        SharedPreferences prefs = getSharedPreferences(GLOBAL_PREFS, MODE_PRIVATE);
+        String username = prefs.getString(MY_EMAIL, "");
+        String password = prefs.getString(MY_PASSWORD,"");
+        editEmail.setText(username);
         Button EditProfile_Button = findViewById(R.id.editprofile_button);
 
 
@@ -37,10 +48,14 @@
                 String NewPassword = editNewPassword.getText().toString();
                 if(!OldPassword.isEmpty())
                 {
-                    if(OldPassword.matches(OldPassword))
+                    if(OldPassword.matches(password))
                     {
                         if(!NewPassword.isEmpty())
                         {
+                            @Override onSuccess(){
+                            editor.putString(MY_EMAIL, email);
+                            editor.putString(MY_PASSWORD, password);
+                        }
                             Toast.makeText(EditProfile.this,"Profile Sucessfully Updated",Toast.LENGTH_SHORT).show();
                             startActivity(new Intent(EditProfile.this,Profile.class));
                             finish();
